pkgs/initiator/initiator.go

```diff
@@ -369,13 +428,87 @@ func (c *Initiator) StartResigning(id [24]byte, ids []uint64, proofs []*spec.Sig
 		KeysharesData: keysharesData,
 		Proofs:        proofsData,
 	}
-	err = c.sendResult(resultMsg, ops, consts.API_RESULTS_URL, id)
+	err = c.sendResult(resultMsg, rMsg.Operators, consts.API_RESULTS_URL, id)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("🤖 Error storing results at operators %w", err)
+	}
+	return depositDataJson, keyshares, proofsArray, nil
+}
+
+func (c *Initiator) StartResharing(id [24]byte, reshareMsg *wire.ReshareMessage) (*wire.DepositDataCLI, *wire.KeySharesCLI, []*wire.SignedProof, error) {
+	dkgResultsBytes, err := c.messageFlowHandlingReshare(id, reshareMsg)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	dkgResults, err := parseDKGResultsFromBytes(dkgResultsBytes, id)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	c.Logger.Info("🏁 DKG Reshare ceremony completed, validating results...")
+	_, _, _, err = spec.ValidateResults(
+		reshareMsg.SignedReshare.Reshare.NewOperators,
+		reshareMsg.SignedReshare.Reshare.WithdrawalCredentials,
+		reshareMsg.Proofs[0].Proof.ValidatorPubKey,
+		reshareMsg.SignedReshare.Reshare.Fork,
+		reshareMsg.SignedReshare.Reshare.Owner,
+		reshareMsg.SignedReshare.Reshare.Nonce,
+		id,
+		dkgResults)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	depositDataJson, keyshares, err := c.processDKGResultResponse(
+		dkgResults,
+		id,
+		reshareMsg.SignedReshare.Reshare.NewOperators,
+		reshareMsg.SignedReshare.Reshare.WithdrawalCredentials,
+		reshareMsg.SignedReshare.Reshare.Fork,
+		reshareMsg.SignedReshare.Reshare.Owner,
+		reshareMsg.SignedReshare.Reshare.Nonce)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if err := crypto.ValidateDepositDataCLI(depositDataJson, common.BytesToAddress(reshareMsg.SignedReshare.Reshare.WithdrawalCredentials)); err != nil {
+		return nil, nil, nil, err
+	}
+	if err := crypto.ValidateKeysharesCLI(keyshares,
+		reshareMsg.SignedReshare.Reshare.NewOperators,
+		reshareMsg.SignedReshare.Reshare.Owner,
+		reshareMsg.SignedReshare.Reshare.Nonce,
+		depositDataJson.PubKey); err != nil {
+		return nil, nil, nil, err
+	}
+	// sending back to operators results
+	depositData, err := json.Marshal(depositDataJson)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	keysharesData, err := json.Marshal(keyshares)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	var proofsArray []*wire.SignedProof
+	for _, res := range dkgResults {
+		proofsArray = append(proofsArray, &wire.SignedProof{res.SignedProof}) //nolint:all
+	}
+	proofsData, err := json.Marshal(proofsArray)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	resultMsg := &wire.ResultData{
+		Operators:     reshareMsg.SignedReshare.Reshare.NewOperators,
+		Identifier:    id,
+		DepositData:   depositData,
+		KeysharesData: keysharesData,
+		Proofs:        proofsData,
+	}
+	err = c.sendResult(resultMsg, reshareMsg.SignedReshare.Reshare.NewOperators, consts.API_RESULTS_URL, id)
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("🤖 Error storing results at operators %w", err)
 	}
```

Comment:
Can wrap these in another function as it is the same for init, resign, reshare (also commented in pull request 100)